fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/MaterializationContext.java

```diff
@@ -78,7 +78,7 @@ public abstract class MaterializationContext {
     // Mark enable record failure detail info or not, because record failure detail info is performance-depleting
     protected final boolean enableRecordFailureDetail;
     // The mv plan struct info
-    protected final StructInfo structInfo;
+    protected StructInfo structInfo;
```

Comment:
why remove final?

Reply:
construct sturctInfo may be failed when materialization init as following:
field `StructInfo` maybe not init, so remove the final
```
 List<StructInfo> viewStructInfos;
        try {
            viewStructInfos = MaterializedViewUtils.extractStructInfo(mvPlan, cascadesContext, new BitSet());
            if (viewStructInfos.size() > 1) {
                // view struct info should only have one, log error and use the first struct info
                LOG.warn(String.format("view strut info is more than one, materialization name is %s, mv plan is %s",
                        getMaterializationQualifier(), getMvPlan().treeString()));
            }
        } catch (Exception exception) {
            LOG.warn(String.format("construct mv struct info fail, materialization name is %s, mv plan is %s",
                    getMaterializationQualifier(), getMvPlan().treeString()), exception);
            this.available = false;
            return;
        }
        this.structInfo = viewStructInfos.get(0);
```

Reply:
set structInfo to null before return in catch?